### PR TITLE
Cache - Deal with signature encoding

### PR DESCRIPTION
**Pull request by keradus**
Discussion about fixing part has taken place in #2463.

### CODE REVIEW

File: tests/Cache/CacheTest.php [hunk 1]
```diff
@@ -174,6 +170,27 @@ public function testCanConvertToAndFromJson()
...
+                array(
+                    'foo' => true,
+                    'bar' => true,
+                )
+            )),
+            array(new Signature(
+                PHP_VERSION,
+                ToolInfo::getVersion(),
+                array(
+                    'header_comment' => array('header' => 'Dariusz '.base64_decode('UnVtafFza2k='))
```
Comment (keradus) [inline tests/Cache/CacheTest.php]: the 2nd part is `Rumiński`, but in ANSI, not UTF.

**Comment by keradus** [comment]
@localheinz, please, could you take a look at this ?

**Comment by keradus** [comment]
@localheinz, please, could you take a look at this ?

File: composer.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
         "symfony/event-dispatcher": "^2.1 || ^3.0",
         "symfony/filesystem": "^2.4 || ^3.0",
         "symfony/finder": "^2.2 || ^3.0",
-        "symfony/polyfill-php54": "^1.0",
+        "symfony/polyfill-php55": "^1.0",
```
Comment (GrahamCampbell) [inline composer.json]: we'll need 5.4 too?

Comment (GrahamCampbell) [inline composer.json]: the 5.5 polyfill does not pull in 5.4

**Comment by GrahamCampbell** [comment]
We probably need to force UTF8 encoding when we encode the json, so decode is straightforward.

**Comment by keradus** [comment]
if you could send a fix basing on my test it would be great

**Comment by localheinz** [comment]
@keradus 

Apologies for the delay!

**Comment by keradus** [comment]
nothing to be sorry about ;)
if @GrahamCampbell has a proposal how to fix it I would appreciate a ready for merge PR ;)

**Comment by GrahamCampbell** [comment]
Is 2.0 a better target for this

**Comment by GrahamCampbell** [comment]
Ok, how does this look: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2463.

**Comment by GrahamCampbell** [comment]
Tests pass on my machine for that other PR.

**Comment by keradus** [comment]
Thank you @GrahamCampbell.

**Pull request merged**